### ISSUE TITLE
TypeError loading empty cell array

### ISSUE THREAD

**Issue by gvnwst** [skjerns/mat7.3#14]
When I attempt to load a fairly vanilla `.mat`, I get a TypeError. By generating some simple test cases with similar datatypes, it appears to be hanging up when I provide an empty matlab cell array. 

### Command 
```
data_dict = mat73.loadmat('test_dataset_nodatetime.mat', use_attrdict=True)
```

### Output
```
TypeError                                 Traceback (most recent call last)
<ipython-input-16-97a7782477dc> in <module>
----> 1 data_dict = mat73.loadmat('test_dataset_nodatetime.mat', use_attrdict=True)

C:\<path>\lib\site-packages\mat73\__init__.py in loadmat(filename, use_attrdict, verbose)
    207     try:
    208         with h5py.File(filename, 'r') as hdf5:
--> 209             dictionary = decoder.mat2dict(hdf5)
    210         return dictionary
    211     except OSError:

C:\<path>\lib\site-packages\mat73\__init__.py in mat2dict(self, hdf5)
     48             ext = os.path.splitext(hdf5.filename)[1].lower()
     49             if ext.lower()=='.mat':
---> 50                 d[var] = self.unpack_mat(hdf5[var])
     51             elif ext=='.h5' or ext=='.hdf5':
     52                 err = 'Can only load .mat. Please use package hdfdict instead'\

C:\<path>\lib\site-packages\mat73\__init__.py in unpack_mat(self, hdf5, depth)
     73                 matlab_class = hdf5[key].attrs.get('MATLAB_class')
     74                 elem   = hdf5[key]
---> 75                 unpacked = self.unpack_mat(elem, depth=depth+1)
     76                 if matlab_class==b'struct' and len(elem)>1:
     77 

C:\<path>\lib\site-packages\mat73\__init__.py in unpack_mat(self, hdf5, depth)
    104             return d
    105         elif isinstance(hdf5, h5py._hl.dataset.Dataset):
--> 106             return self.convert_mat(hdf5)
    107         else:
    108             raise Exception(f'Unknown hdf5 type: {key}:{type(hdf5)}')

C:\<path>\lib\site-packages\mat73\__init__.py in convert_mat(self, dataset)
    140             for ref in dataset:
    141                 row = []
--> 142                 for r in ref:
    143                     entry = self.unpack_mat(self.refs.get(r))
    144                     row.append(entry)

TypeError: 'numpy.uint64' object is not iterable

```

The behavior is replicated with or without the `use_attrdict` flag.
I can provide the file if that's helpful.

**Comment by skjerns** [comment]
Yes, please provide the simple test file or test case.

If you feel comfortable you can also directly provide a PR with a fix and test for this case.

**Comment by gvnwst** [comment]
I'm fairly new to Python so I'll take a look at it, but don't hold your breath.

Here's a `.mat` with two variables: an empty cell array and a simple double array.
[empty_cell_array.zip](https://github.com/skjerns/mat7.3/files/6554895/empty_cell_array.zip)

**Comment by skjerns** [comment]
By the way, I just tried the file you supplied, and I could load it without problems. Are you sure this problem persists with the most recent version of numpy and mat73?

**Comment by skjerns** [comment]
I fixed the error in v0.49 :) please test the latest version, should work now.

**Comment by gvnwst** [comment]
Hmm based on the files in my environment's `site-packages`, it looks like I had `v0.46`. I don't recall whether I used `pip install` or cloned the repo and installed manually. 
My numpy version is 1.20.0 -- not the latest, but reasonably up-to-date. 

I just pulled the updates and installed manually, it now works! I get it to import as an empty array.
A small suggestion might be to import cell arrays as lists instead of `numpy` arrays -- I'm not a super experienced user, but these seem like the most similar datatypes (in that matlab cell arrays can hold any other type of matlab object, and each entry of the cell need not be the same object class).
